docs/index.rst

```diff
@@ -37,7 +37,7 @@ range of statistics and robust optimisers.
 Sherpa is released under the
 `GNU General Public License v3.0
 <https://github.com/sherpa/sherpa/blob/master/LICENSE>`_,
-and is compatible with Python versions 3.8 to 3.10.
+and is compatible with Python versions 3.8 to 3.11.
```

Comment:
Do we want to go up to 3.9 as the minimum while we are at it? I know we have separate PRs for that, but it really does not hurt here - we don't *promise* that it fails below 3.9 and if we do it, we won't forget!

Reply:
I decide to just add 3.11 here and worry about dropping 3.8 in a different PR (as we don't make use of any 3.9-only functionality).